lib/licensee/matchers/nuget.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+module Licensee
+  module Matchers
+    class NuGet < Licensee::Matchers::Package
+      # While we could parse the nuspec file, prefer a lenient regex for speed and security.
+      # Moar parsing moar problems.
+      LICENSE_REGEX = %r{
+        <license\s*type\s*=\s*["']expression["']\s*>([a-z\-0-9\. +()]+)<\/license\s*>
```

Comment:
Would it be worth also matching licenseUrl where the URL likely contains a SPDX ID in the path which can be extracted?

opensource.org/licenses/{id} seems [pretty common](https://github.com/search?q=filename%3A.nuspec+licenseurl+opensource.org&type=Cod). spdx.org/licenses and licenses.nuget.org values exist but are surprisingly uncommon.

Not essential, just a thought.

Reply:
I think it would be useful for opensource.org, and perhaps others like https://www.apache.org/licenses/LICENSE-2.0.txt that I've noticed a fair number of packages use. I agree with your assessment that licenses.nuget.org isn't especially useful - most of the time it's used, there's a `<license>` expression anyways since those features were added to nuget around the same time.

But there's arguably a question of ownership between `licensee/licensee` and `github/licensed` when it comes to `licenseUrl`. My understanding is that fetching and matching the content of the URL (when it isn't well-known) might not meet the goals of `licensee` - I don't think any other project type in `licensee` currently downloads license content from the internet, whereas `licensed` already does that for some package types.

It seemed difficult to split it between the two, since `licensed` would presumably want to know if the match for `package_file` was due to a `license` or `licenseUrl`, so it could avoid downloading the license if it was well-known, but still download it in the case where `licensee` found a `license` but `licenseUrl` was a non-well-known URL. package.rb doesn't currently have a mechanism to expose something like that.

Reply:
Fetching the content specified by `licenseUrl` would be out of scope for licensee IMO, but extracting license identification from well known URLs seems within scope for licensee.

Licensed _may_ wish to do a bit of duplicate work to avoid downloading well known license texts unnecessarily.

Reply:
> Fetching the content specified by licenseUrl would be out of scope for licensee IMO, but extracting license identification from well known URLs seems within scope for licensee.

I'm decidedly unopinionated on what logic should exist here vs licensed and am happy to follow opinion on what's decided here, but this generally makes sense to me 👍 

> Licensed may wish to do a bit of duplicate work to avoid downloading well known license texts unnecessarily.

👍 licensed already autogenerates text when a definitive classification is made but no license text is available from a known source.  It shouldn't be too hard to extend that to a scenario where there is 100% confidence in the license classification and a costly operation to get license text would match autogenerated text

Reply:
Cool, I'll move that logic into `licensee` then and update this PR.

Reply:
Added matching for `licenseUrl` to recognize licenses.nuget.org, opensource.org, spdx.org, and apache.org well-known licenses. Added relevant tests.